### ISSUE TITLE
return error codes and deal with missing variable names

### ISSUE THREAD

**Issue by ecoon** [amanzi/amanzi#801]
HDF5_MPI currently has no error handling, really because ASCEMIO has no error handling.  If a user causes the code to ask for an attribute or field that does not exist in an HDF5 file, HDF5 prints a bunch of stuff somewhere but there is no error code, no thrown error, etc.  So there is no way for Amanzi to detect that HDF5 was unable to find the thing, and therefore no way to throw a coherent error message.

This could be fixed by either fixing ASCEMIO or removing it entirely.

**Comment by ecoon** [comment]
@lipnikov  is this related to #799 ?

**Comment by lipnikov** [comment]
@ecoon It is tangent. In #799, the name of a field was mis-formed inside C++ code, but the initialize-from-file scenario does not require that a field be present. In other words, this is a partial initialization of state.

**Comment by lipnikov** [comment]
@ecoon is this related to #76? If so, can we close the old ticket?

**Comment by ecoon** [comment]
Ok, this really bit Bo and me recently -- in a big parallel intermediate scale model there was a bug in the attribute names, which were then being read from the incorrect checkpoint file, which resulted in... no errors because ascemio doesn't deal with errors.  This is really just plain stupid.

I have a fix for ascemio to at least error when reading attributes.  I may fix it for vectors as well.  How are we dealing with ascemio?  Is there a repo somewhere that I can push code to?  Or do I just add more patches to the TPLs?

**Comment by ecoon** [comment]
> @ecoon is this related to #76? If so, can we close the old ticket?

yes, I closed that ticket as this one is "bigger" than that one and fixing this will fix that.
